tests/test_utils.py

```diff
@@ -259,16 +259,17 @@ def test_try_acquire():
 async def test_concurrent_runner():
     results = []
 
-    def _results_callback(result):
-        results.append(result)
+    def _results_callback(task):
+        results.append(task.result())
```

Comment:
Should we drop all these tests that were relying on callbacks?

Reply:
no, because they're actually testing that functionality in ConcurrentTasks is working. We could re-write them to have the `coroutine` functions directly modify the `results` variables, but I thought this would be easier to review since it wasn't much rewriting of logic.